src/devdocs2zim/generator.py

```diff
@@ -491,6 +516,39 @@ def generate_zim(
             )
         return zim_path
 
+    @staticmethod
+    def fetch_logo_bytes(user_logo_path: str) -> bytes:
```

Comment:
I don't feel like it is necessary to operate on real files, it is not like we expect the image to be very very big, it should fit into memory and avoid having to mess with filenames and temporary stuff to cleanup afterwards. At least experienced showed us it is preferable to avoid messing with the filesystem whenever possible. All zimscraperlib methods are capable to operate on in-memory `BytesIO`.

You can get inspiration (not necessarily the best implementation ever, not at all) at https://github.com/openzim/mindtouch/blob/5d68d9a22fbdac4d88dd8b2f774fd5cc47be3a13/scraper/src/mindtouch2zim/processor.py#L563 ; this implementation probably only supports online URLs, at least this was the use case I had in mind when developing it.

Reply:
Moved us to using byte buffers.